### PR TITLE
Follow redirection responses when refreshing stale cache objects.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci clang-format]

**Review by randall — APPROVED** [review]

**Comment by randall** [comment]
Thanks for this!

**Comment by ywkaras** [comment]
Fortunately turned out to be a simple fix.

File: tests/gold_tests/redirect/redirect_stale.test.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    'body': ''}
+server.addResponse('sessionfile.log', request_header, response_header)
+
+ts = Test.MakeATSProcess("ts")
+
+ts.Disk.records_config.update({
+    'proxy.config.diags.debug.enabled': 1,
+    'proxy.config.diags.debug.tags': 'http|dns|cache|redirect',
+    'proxy.config.http.cache.required_headers': 0,  # Only Content-Length header required for caching.
+    'proxy.config.http.push_method_enabled': 1,
```
Comment (ywkaras) [inline tests/gold_tests/redirect/redirect_stale.test.py:60]: Enabling the push method is kruft, this was accidentally copied from another Au test.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**